CONTRIBUTING.md

```diff
@@ -0,0 +1,263 @@
+    *   If you've introduced some new feature, write about who might use it,
+        give a breif code sample to show it and let us know how you tested it!
+    *   If you've fixed a bug, write about why this bug exists in the first place,
+        how you solved it and how a test makes sure it won't pop up again!
+
+*   Once you've submitted a PR, we'll probably enable some automatic tests to run.
+    This will make sure all the tests run smoothly, make sure the documentation builds correctly and do some quick check on code quality.
+    You'll be able to see these run in the **Checks** tab or at the bottom of the PR.
+    If you see a red x, that means somethings probably gone wrong which should have been caught by running the tests locally but we also do some checks in environments you were not developing on.
+    Sometimes it is the case that there is a bug unrelated to your changes that cause the tests to fail but we are aware of these.
+    If you see one of these failures, feel free to ask!
+
+*   Meanwhile, we'll also review your code.
+    In the top right you'll see a little **Suggested Reviewers** button so feel free to request a review with that!
+    Some common review points are:
+    * This seems odd, why was this done?
+    * Could you see if you can use the functionality from place X?
+    * Could you creat a test or documentation for this?
+    * This could be a nicer way of doing this, what do you think?
+    Occasionally there will be some major point which will require more discussion but those are more on a case-by-case basis.
+
+*   This process of review, fix and testing may go on a few times.
+    The simplest way to reduce the time needed and to help us too is to run the tests, code formatting check and doc building locall.
+    If they all pass locally they will very often have no issues in the automated tests.
+
+    Once everyone is happy, it's time for us to hit (*squash*) merge, get it into the development branch and have one more contribution to autosklearn!
+
+## Kinds of PRs
+Of course we can't cover every use case but here's some common kinds of PR's along with some pointers!
+
+#### Documentation
+Anything to contribute to better documentation is always appreciated
+#### Bug Fixes
+* TODO
+#### Features
+* TODO
+
+# General FAQ
+
+### I've finished my pull request or made new changes and want it reviewed, now what?
+* In the Pull request on the right there should be a tab **Reviewers** with some suggested
+reviewers, feel free to request once you think your contribution is ready.
+
+### I've been asked to rebase my pull request, why and what do I do?
+* It can often be the case that while you were working on some changes, we may have merged something new into the development branch.
+This is a problem because the branch you created was based off the old development branch in and so there are no changes in the automl/auto-sklearn code base you don't have in your branch.
```

Comment:
```suggestion
This can be a problem because the branch you created was based off the old development branch in and so there are new changes in the automl/auto-sklearn code base you don't have in your branch.
```